handshake_ix.go

```diff
@@ -23,13 +23,17 @@ func ixHandshakeStage0(f *Interface, hh *HandshakeHostInfo) bool {
 		return false
 	}
 
-	// If we're connecting to a v6 address we must use a v2 cert
 	cs := f.pki.getCertState()
 	v := cs.initiatingVersion
-	for _, a := range hh.hostinfo.vpnAddrs {
-		if a.Is6() {
-			v = cert.Version2
-			break
+	if hh.initiatingVersionOverride != cert.VersionPre1 {
+		v = hh.initiatingVersionOverride
+	} else if v < cert.Version2 {
+		// If we're connecting to a v6 address we must use a v2 cert
```

Comment:
I think the `must` here is really a `should try super hard but maybe its ok`

Reply:
I know it's just a comment but in the spirit of #1512, it would be nice to reflect that intent here ("try to use a v2 cert for this" type of thing)